nautobot/dcim/models/devices.py

```diff
@@ -1220,3 +1230,144 @@ def __str__(self):
         if self.alias:
             return self.alias
         return f"{self.platform} - {self.version}"
+
+
+#
+# Controller
+#
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "locations",
+    "statuses",
+    "webhooks",
+)
+class Controller(PrimaryModel, ConfigContextModel):
+    """Represents an entity that manages or controls one or more devices, acting as a central point of control.
+
+    A Controller can be deployed to a single device or a group of devices represented by a DeviceRedundancyGroup.
+    """
+
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
+    status = StatusField(blank=False, null=False)
+    description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
+    location = models.ForeignKey(
+        to="dcim.Location",
+        on_delete=models.PROTECT,
+        related_name="controllers",
+    )
+    platform = models.ForeignKey(
+        to="dcim.Platform",
+        on_delete=models.SET_NULL,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+    role = RoleField(blank=False, null=False)
+    tenant = models.ForeignKey(
+        to="tenancy.Tenant",
+        on_delete=models.PROTECT,
+        related_name="controllers",
+        blank=True,
+        null=True,
+    )
+    external_integration = models.ForeignKey(
+        to="extras.ExternalIntegration",
+        on_delete=models.SET_NULL,
```

Comment:
This "feels" like it should be `PROTECT` - I don't think it makes sense to allow a user to delete an ExternalIntegration if there's a Controller using it.

Reply:
WDYT @whitej6 ?